src/Authenticator/CookieAuthenticator.php

```diff
@@ -154,7 +156,14 @@ protected function _createPlainToken($identity): string
         $usernameField = $this->getConfig('fields.username');
         $passwordField = $this->getConfig('fields.password');
 
-        return $identity[$usernameField] . $identity[$passwordField];
+        if (!$this->getConfig('useSalt')) {
+            return $identity[$usernameField] . $identity[$passwordField];
+        }
+
+        // Prevent user from unknowingly using an empty salt
+        $salt = Configure::readOrFail('Security.salt');
```

Comment:
How about also allowing to explicitly set a different salt through the authenticator config?

Reply:
Other authenticators support it, so I'd say yes.

Also this needs to use the `Security` class, as that config setting is by default being consumed in applications base on the app skeleton. And since there hasn't been a move yet to tie the plugin harder to the core, it should then also not be a hard dependency.

https://github.com/cakephp/authentication/blob/master/src/Authenticator/HttpDigestAuthenticator.php#L60

Reply:
I'll change it to the Security utility.

About user defined salt, shall we make the config option called 'salt' and use it as:

- true: Security salt
- false: No salt
- string: user-defined salt

Is that intuitive?